src/tools/pbs_upgrade_job.c

```diff
@@ -194,224 +286,258 @@ check_job_file(int fd)
-	tmpfd = fileno(tmp);
-	if (tmpfd < 0) {
-		fprintf(stderr, "Failed to find temporary file descriptor [%s]\n",
-			errno ? strerror(errno) : "No error");
-		return 1;
-	}
-
-	/* Write the new jobfix structure to the output file */
-	len = write(tmpfd, &new_job.ji_qs, sizeof(new_job.ji_qs));
-	if (len != sizeof(new_job.ji_qs)) {
-		fprintf(stderr, "Failed to write jobfix to output file [%s]\n",
-			errno ? strerror(errno) : "No error");
-		return 1;
-	}
-
-	/* Read the rest of the input and write it to the temporary file */
-	do {
-		len = read(fd, buf, sizeof(buf));
+	if (jobfix_version == 12) {
+		memset(&old_jobfix_v12, 0, sizeof(old_jobfix_v12));
+		len = read(fd, (char *)&old_jobfix_v12, sizeof(old_jobfix_v12));
 		if (len < 0) {
 			fprintf(stderr, "Failed to read input file [%s]\n",
-				errno ? strerror(errno) : "No error");
+					errno ? strerror(errno) : "No error");
 			return 1;
 		}
-		if (len < 1)
-			break;
-		len = write(tmpfd, buf, len);
-		if (len < 0) {
-			fprintf(stderr, "Failed to write output file [%s]\n",
-				errno ? strerror(errno) : "No error");
+		if (len != sizeof(old_jobfix_v12)) {
+			fprintf(stderr, "Format not recognized, not enough fixed data.\n");
+			return 1;
+		}
+		if (old_jobfix_v12.ji_jsversion != JSVERSION_18) {
+			fprintf(stderr, "Job format version not recognized.\n");
 			return 1;
 		}
-	} while (len > 0);
-
-	/* Reset the file descriptors to zero */
-	pos = lseek(fd, 0, SEEK_SET);
-	if (pos != 0) {
-		fprintf(stderr, "Failed to reset job file position [%s]\n",
-			errno ? strerror(errno) : "No error");
-		return 1;
-	}
-	pos = lseek(tmpfd, 0, SEEK_SET);
-	if (pos != 0) {
-		fprintf(stderr, "Failed to reset temporary file position [%s]\n",
-			errno ? strerror(errno) : "No error");
-		return 1;
-	}
 
-	/* Copy the data from the temporary file back to the original */
-	do {
-		len = read(tmpfd, buf, sizeof(buf));
+		/* Copy the data to the new jobfix structure */
+		memset(&new_job, 0, sizeof(new_job));
+		memcpy(&new_job.ji_qs, &old_jobfix_v12, sizeof(old_jobfix_v12));
+		snprintf(new_job.ji_qs.ji_jobid, sizeof(new_job.ji_qs.ji_jobid),
+				"%s", old_jobfix_v12.ji_jobid);
+		snprintf(new_job.ji_qs.ji_fileprefix, sizeof(new_job.ji_qs.ji_fileprefix),
+				"%s", old_jobfix_v12.ji_fileprefix);
+		snprintf(new_job.ji_qs.ji_queue, sizeof(new_job.ji_qs.ji_queue),
+				"%s", old_jobfix_v12.ji_queue);
+		snprintf(new_job.ji_qs.ji_destin, sizeof(new_job.ji_qs.ji_destin),
+				"%s", old_jobfix_v12.ji_destin);
+		new_job.ji_qs.ji_un_type = old_jobfix_v12.ji_un_type;
+		memcpy(&new_job.ji_qs.ji_un, &old_jobfix_v12.ji_un, sizeof(new_job.ji_qs.ji_un));
+
+	} else {
+
+		/* Read in the pre19 jobfix structure */
+		memset(&old_jobfix_pre19, 0, sizeof(old_jobfix_pre19));
+		len = read(fd, (char *)&old_jobfix_pre19, sizeof(old_jobfix_pre19));
 		if (len < 0) {
-			fprintf(stderr, "Failed to read temporary file [%s]\n",
-				errno ? strerror(errno) : "No error");
+			fprintf(stderr, "Failed to read input file [%s]\n",
```

Comment:
Is there any reason to change the comment to use input file instead of temporary file ?